controllers/provisionserver_controller.go

```diff
@@ -0,0 +1,310 @@
+	provisionserver "github.com/openstack-k8s-operators/osp-director-operator/pkg/provisionserver"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+// ProvisionServerReconciler reconciles a ProvisionServer object
+type ProvisionServerReconciler struct {
+	client.Client
+	Kclient kubernetes.Interface
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=provisionservers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=provisionservers/finalizers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=provisionservers/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;create;update;delete;watch;
+// +kubebuilder:rbac:groups=core,resources=configmaps/finalizers,verbs=get;list;create;update;delete;watch;
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;create;update;delete;watch;
+// +kubebuilder:rbac:groups=core,resources=volumes,verbs=get;list;create;update;delete;watch;
+// +kubebuilder:rbac:groups=core,resources=nodes,verbs=get;list;update;watch;
+// +kubebuilder:rbac:groups=core,resources=pods,verbs=get;list;update;watch;
+// +kubebuilder:rbac:groups=machine.openshift.io,resources="*",verbs="*"
+// +kubebuilder:rbac:groups=metal3.io,resources="*",verbs="*"
+// +kubebuilder:rbac:groups=security.openshift.io,namespace=openstack,resources="securitycontextconstraints",resourceNames="privileged",verbs="use"
+// +kubebuilder:rbac:groups=security.openshift.io,namespace=openstack,resources="securitycontextconstraints",resourceNames="anyuid",verbs="use"
+
+// Reconcile - provision image servers
+func (r *ProvisionServerReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	_ = r.Log.WithValues("provisionserver", req.NamespacedName)
+
+	// Fetch the ProvisionServer instance
+	instance := &ospdirectorv1beta1.ProvisionServer{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers. Return and don't requeue.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// config maps
+	op, err := r.httpdConfigMapCreateOrUpdate(instance)
+
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if op != controllerutil.OperationResultNone {
+		r.Log.Info(fmt.Sprintf("Httpd ConfigMap %s successfully reconciled - operation: %s", instance.Name, string(op)))
+	}
+
```

Comment:
I think we could do all the configmap creation from the templates using template like https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/controllers/controllervm_controller.go#L124-L147 and use common.EnsureConfigMaps https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/pkg/common/configmap.go#L107 . but we can test/do that in a follow up. Note: when I did this for secrets like the controllervm use it, I added the same for configmaps, but it is not yet used and tested, but its how I did it for the e.g. nova-operator

Reply:
Sounds good to me.  This was just a quick approach along the "fail fast" track to just see if all of this would even work.